Program.cs

```diff
@@ -11,101 +49,96 @@ static Rectangle CellToRectangle(int row, int col, int size)
     }
 
     // trying to draw the apple shape? 
-    static Vector2 CellToCenter(int row, int col, int size)
+    static Vector2 CellToCenter(Cord appleCord, int size)
     {
         // Uhhh, we're converting thaaa indices of the cells into 
         // the centers of where they need to drawn (in the pixel world :o )
-        float centerX = size * (col + 0.5f);
-        float centerY = size * (row + 0.5f);
+        float centerX = size * (appleCord.Y + 0.5f);
+        float centerY = size * (appleCord.X + 0.5f);
         return new Vector2(centerX, centerY);
     }
 
-    static void DrawGrid(int rows, int cols, int size)
+    static void DrawGrid(Grid grid, int size)
     {
-        for (int row = 0; row < rows; row += 1)
+        for (int row = 0; row < grid.cols; row += 1)
```

Comment:
again gurl, you gotta sort this out. you are looping through rows, i.e. Y coordinate but you are stopping at the cols. If you have an asymmetric grid this breaks.